app/services/requirements/schedule_checker.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module Requirements
+  class ScheduleChecker
+    include ActionView::Helpers::DateHelper
+
+    attr_reader :revision
+
+    MAX_PUBLISH_DELAY = 14.months
+    MIN_PUBLISH_DELAY = 15.minutes
+
+    def initialize(revision)
+      @revision = revision
+    end
+
+    def pre_schedule_issues
+      issues = []
+
+      if publish_at > MAX_PUBLISH_DELAY.from_now
+        issues << Issue.new(:schedule_date,
+                            :too_far_in_future,
+                            time_period: distance_of_time_in_words(MAX_PUBLISH_DELAY))
```

Comment:
Is this where we get "about a year" ?

I don't really think we should use the `distance_of_time_in_words` helper as any changes to the constants may make it's outputs rather confusing which then means we may as well have hardcoded the value as that'd at least be clearer that something had to change with the number.

Reply:
Agreed, that did annoy me in the tests, although I would prefer not to go back to the original either, as it was somewhat confusing to me. Fortunately, I think I found an alternative - see what you think :-).

Reply:
That seems good 👍 are they covered by a test? 

My only concern would be that we're relying on the ActiveSupport::Duration#inspect method in a way that it's not really intended for, but as long as we have a test that blows up in the very unlikely even that it changes I'm happy. It's a shame the `to_s` method isn't set up to do this as that feels far more natural than it's string integer return 🤔

Reply:
OK. I've specified the periods explicitly in the test - I forgot to change one of them anyway. That OK?

Reply:
Sounds good